cypress/support/pages/signUp.pageObject.js

```diff
@@ -0,0 +1,85 @@
+import PageObject from '../PageObject';
+
+class SignUpPageObject extends PageObject {
+  url = '/#/register';
+
+  get emailField() {
+    return cy.getByDataQa('email-sign-up');
+  }
+
+  get passwordField() {
+    return cy.getByDataQa('password-sign-up');
+  }
+
+  get usernameField() {
+    return cy.getByDataQa('username-sign-up');
+  }
+
+  get signUpBtn() {
+    return cy.getByDataQa('sign-up-btn');
+  }
+
+  typeEmail(email) {
+    this.emailField
+      .type(email);
+  }
+
+  typePassword(password) {
+    this.passwordField
+      .type(password);
+  }
+
+  typeUsername(username) {
+    this.usernameField
+      .type(username);
+  }
+
+  clickSignUpBtn() {
+    this.signUpBtn
+      .click();
+  }
+
+  assertUsernameRequiredModal() {
+    cy.contains('.swal-modal', 'Username field required.')
+      .should('be.visible');
+  }
+
+  assertPasswordRequiredModal() {
+    cy.contains('.swal-modal', 'Password field required.')
+      .should('be.visible');
+  }
+
+  assertEmailRequiredModal() {
+    cy.contains('.swal-modal', 'Email field required.')
+      .should('be.visible');
+  }
+
+  assertSuccessfulRegistrationModal() {
+    cy.contains('.swal-modal', 'Your registration was successful!')
+      .should('be.visible');
+  }
+
+  assertEmailTakenModal() {
+    cy.contains('.swal-modal', 'Email already taken.')
+      .should('be.visible');
+  }
```

Comment:
those methods are duplicates of each other.You could make 1 method like assertModalError or something like that that takes string and checks it
```suggestion
  assertUsernameRequiredModal() {
    cy.contains('.swal-modal', 'Username field required.')
      .should('be.visible');
  }

  assertPasswordRequiredModal() {
    cy.contains('.swal-modal', 'Password field required.')
      .should('be.visible');
  }

  assertEmailRequiredModal() {
    cy.contains('.swal-modal', 'Email field required.')
      .should('be.visible');
  }

  assertSuccessfulRegistrationModal() {
    cy.contains('.swal-modal', 'Your registration was successful!')
      .should('be.visible');
  }

  assertEmailTakenModal() {
    cy.contains('.swal-modal', 'Email already taken.')
      .should('be.visible');
  }
```